crates/metaslang/cst/src/cursor.rs

```diff
@@ -6,10 +6,34 @@ use crate::kinds::KindTypes;
 use crate::nodes::{Edge, Node, NonterminalNode};
 use crate::text_index::{TextIndex, TextRange};
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+enum Parent<T: KindTypes> {
```

Comment:
rather than introducing the new `Parent` abstraction, which can introduce subtle bugs in the future, if other `Closed` properties are used/modified, WDYT of restricting it in the type-system itself?

```rust
enum PathAncestor<T: KindTypes> {
  None,
  Disconnected {
    label: EdgeLabel,
  },
  Connected {
    parent: Rc<PathAncestor<T>>,
    nonterminal_node: Rc<NonterminalNode<T>>,
    child_number: usize,
    text_offset: TextIndex,
  }
}
```